components/style/gecko/wrapper.rs

```diff
@@ -1136,6 +1146,68 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
         None
     }
 
+    fn attr_matches(&self,
+                    ns: &NamespaceConstraint<&Namespace>,
+                    local_name: &Atom,
+                    operation: &AttrSelectorOperation<&Atom>)
+                    -> bool {
+        unsafe {
+            match *operation {
+                AttrSelectorOperation::Exists => {
```

Comment:
These two large `match` expressions end up calling different sets of FFI functions. So unless macros are involved, shared code would requires passing a number of dynamically-dispatched function pointers or something.